Update changelog with notes for v1.4.0 (#3315)
spiffe/spire: CHANGELOG.md

# Changelog

## [1.4.0] - 2022-08-08

### Added 

> - Support K8S worklod attestor using windows containers (#3191)

Reviewer: ```suggestion
- Support for Windows workload attestation on Kubernetes (#3191)
```

> - Support custom key types used for Workload SVIDs (#3237)

Reviewer: ```suggestion
- Support for using RSA keys with Workload X509-SVIDs (#3237)
```

> - Support anonymous authentication on k8s WorkloadAttestor (#3273)

Reviewer: ```suggestion
- Support for anonymous authentication with the Kubelet when performing workload attestation on Kubernetes (#3273)
```

> - Support for anonymous authentication with the Kubelet when performing workload attestation on Kubernetes (#3273)

Reviewer 2: ```suggestion
- Support for anonymous authentication to the Kubelet secure port when performing workload attestation on Kubernetes (#3273)
```

### Deprecated
- The Node Resolver plugin type (#3272)

### Fixed

> - Ensures the can_reattest flag is propogated when the attested node record is updated (#3292)

Reviewer: ```suggestion
- Persistence of the can_reattest flag during agent SVID renewal (#3292)
```

> - A regression in Server API authentication that prevents Agents to be notified when it was evicted (#3269)

Reviewer: ```suggestion
- A regression in behavior preventing an agent from re-attesting when it has been evicted (#3269)
```

### Changed

> - The server's Azure node attestor subsumed all the functionality of the node resolver, which has been deprecated (#3272)

Reviewer: ```suggestion
- The Azure Node Attestor to optionally provide selectors (#3272)
```

> - Fail when prividing unknown configurations to Docker Workload Attestor (#3243)

Reviewer: ```suggestion
- The Docker Workload Attestor now fails when configured with unknown options (#3243)
```

> - Improved CRI-O support with Kubernetes workload attestation (#3183)

Reviewer 2: ```suggestion
- Improved CRI-O support with Kubernetes workload attestation (#3242)
```

We usually link the PRs here, not the issues

> - Add special case on K8S Workload attestor when CRI-O does not provide POD ID on cgroup (#3183)

Reviewer: ```suggestion
- Improved CRI-O support with Kubernetes workload attestation (#3183)
```

> - Agent data storage has been consolidated to a single JSON file (#3201)

Reviewer 2: ```suggestion
- Agent data stored on disk has been consolidated to a single JSON file (#3201)
```
nit: clarifies that this is only affecting the existing files managed on-disk and not data cached in-memory

> - Consolidate agent storage information into a single JSON file (#3201)

Reviewer: ```suggestion
- Agent data storage has been consolidated to a single JSON file (#3201)
```

> - Agent and server data directories on Windows no longer inherited permission from parent directory (#3227)

Reviewer: ```suggestion
- Agent and server data directories on Windows no longer inherit permissions from parent directory (#3227)
```

- Endpoints exposed using named pipes explicitly deny access to remote callers (#3236)
- Small documentation improvements (#3264)

### Removed

> - Remove webhood mode from k8s-workload-registrar (#3235)

Reviewer: ```suggestion
- The deprecated webhook mode from the k8s-workload-registrar (#3235)
```

> - The configmap leader election lock is not longer supported (#3241)

Reviewer: ```suggestion
- Support for the configmap leader election lock type from the k8s-workload-registrar (#3241)
```
